hera_cal/tests/test_abscal.py

```diff
@@ -131,10 +145,78 @@ def test_gains2calfits(self):
+        nt.assert_equal(d[(24,37,'xx')].shape, (60, 64))
+        nt.assert_equal(d[(24,37,'xx')].dtype, np.complex)
+        # test w/ noise
+        d = hc.abscal.wiener(self.data, window=(5, 15), noise=0.1, medfilt=True, medfilt_kernel=(1, 13))
+        nt.assert_equal(d[(24,37,'xx')].shape, (60, 64))
+        # test w/o medfilt
+        d = hc.abscal.wiener(self.data, window=(5, 15), medfilt=False)
+        nt.assert_equal(d[(24,37,'xx')].shape, (60, 64))
+        # test as array
+        d = hc.abscal.wiener(self.data[(24,37, 'xx')], window=(5, 15), medfilt=False, array=True)
+        nt.assert_equal(d.shape, (60, 64))
+        nt.assert_equal(d.dtype, np.complex)
+
+    def test_Baseline(self):
+        bls = map(lambda k: hc.abscal.Baseline(self.antpos[k[1]] - self.antpos[k[0]], tol=2.0), self.data.keys())
+        bls_conj = map(lambda k: hc.abscal.Baseline(self.antpos[k[0]] - self.antpos[k[1]], tol=2.0), self.data.keys())
+        nt.assert_equal(bls[0], bls[0])
+        nt.assert_false(bls[0] == bls[1])
+        nt.assert_equal(bls[0] == bls_conj[0], 'conjugated')
+
+    def test_match_red_baselines(self):
+        model = copy.deepcopy(self.data)
+        model = odict([((k[0]+1, k[1]+1, k[2]), model[k]) for i,k in enumerate(model.keys())])
+        model_antpos = odict([(k+1, self.antpos[k]) for i,k in enumerate(self.antpos.keys())])
+        data = hc.abscal.match_red_baselines(self.data, self.antpos, model, model_antpos, tol=2.0, verbose=False)
+        nt.assert_equal(len(data.keys()), 9)
+        nt.assert_true((25, 38, 'xx') in data)
+
+    def test_mirror_data_to_red_bls(self):
+        # make fake data
+        reds = hc.redcal.get_reds(self.antpos, pols=['xx'])
+        data = DataContainer(odict(map(lambda k: (k[0], self.data[k[0]]), reds[:5])))
+        # test execuation
+        d = hc.abscal.mirror_data_to_red_bls(data, self.antpos)
+        nt.assert_equal(len(d.keys()), 16)
+        nt.assert_true((24, 25, 'xx') in d)
+        # test reweighting
+        w = hc.abscal.mirror_data_to_red_bls(self.wgts, self.antpos, weights=True)
+        nt.assert_equal(w[(24, 25, 'xx')].dtype, np.float)
+        nt.assert_almost_equal(w[(24, 25, 'xx')].max(), 16.0)
+
     def test_echo(self):
         hc.abscal.echo('hi', verbose=True)
         hc.abscal.echo('hi', type=1, verbose=True)
 
+    def test_flatten(self):
+        l = hc.abscal.flatten([['hi']])
+        nt.assert_equal(np.array(l).ndim, 1)
+
+    def test_avg_data_across_red_bls(self):
+        data, flags, antpos, ants, freqs, times, lsts, pols = hc.abscal.UVData2AbsCalDict(self.data_file, return_meta=True)
+        rd, rf, rk = hc.abscal.avg_data_across_red_bls(data, antpos, flags=self.wgts, tol=2.0)
+        rd, rf, rk = hc.abscal.avg_data_across_red_bls(data, antpos, tol=2.0, median=True)
+        rd, rf, rk = hc.abscal.avg_data_across_red_bls(data, antpos, tol=2.0, broadcast_flags=True)
+        nt.assert_equal(len(rd.keys()), 9)
```

Comment:
Again, I think you you need to test a few visibilities to see if this worked properly.

Reply:
yup